lab5.py

```diff
@@ -0,0 +1,103 @@
+from enum import Enum
+
+class City(Enum):
+    LVIV = 1
+    KYIV = 2 
+    BURSHTYN = 3
+
+class WeatherType(Enum):
+    SUNNY = 0
+    CLOUDY = 1
+    RAINY = 2
+    FOGGY = 3
+
+class Weather(object):
+
+    def __init__(self, day, city, country, temp, humidity, wind_speed, weather_type):
+        self.day = day
+        self.city = city
+        self.country = country
+        self.temp = temp
+        self.humidity = humidity
+        self.wind_speed = wind_speed
+        self.weather_type = weather_type
+        
+    def __del__(self):
+        print("Cleanup has been completed")
+        
+    def getAll(self):
+        """
+        Explanatiom huy na blude
+        :return: pizdec
+        """
+        return f"The temperature on {self.day} in {self.city.name}, {self.country} was {self.temp} Celsius degrees. The humidity was {self.humidity} %, the wind speed was {self.wind_speed} and the weather outside is {self.weather_type.name}" 
+
+
+
+class WeatherCalendar(Weather):
+    def __init__(self):
+        self.weather_objects = []
+
+    def add_weather_objects(self, obj):
+        self.weather_objects.append(obj)
+
+    def findMaxTemperature(self):
+        max_temp = None
+        max_temp_day = None
+        for obj in self.weather_objects:
+             if max_temp is None or obj.temp > max_temp:
+                  max_temp = obj.temp
+                  max_temp_day = obj.day
+        if max_temp_day is not None:
+            return max_temp_day
+        else:
+            return "Not enough data"
+        
+    
+    def isLvivWeather(self, humidity, weather_type):
+        for obj in self.weather_objects:
+            if humidity > 80 and weather_type == WeatherType.RAINY and obj.city is City.LVIV:
+                return "The typical day in Lviv"
+            else:
+                return "You're lucky, man"
+
+        
+    def sort_weather(self):
+        self.weather_objects.sort(key=lambda x: x.day)
+
+        
+def main():
+    weather1 = Weather(1, City.LVIV, "Ukraine", 2, 80, 10, WeatherType.FOGGY)
+    weather2 = Weather(9, City.KYIV, "Ukraine", 4, 67, 7, WeatherType.CLOUDY)
+    weather3 = Weather(3, City.BURSHTYN, "Ukraine", 3, 72, 8, WeatherType.SUNNY)
+    weather4 = Weather(6, City.LVIV, "Ukraine", 7, 89, 9, WeatherType.RAINY)

```

Comment:
It is better to change the city in the weather4 to another city

Reply:
I didn't commit your suggestion since I must have an example of the outcome "The typical weather in Lviv", but I added new city Dnipro to another weather instance.